### ISSUE TITLE
top_n_words doesn't work on zero-shot modeling 

### ISSUE THREAD

**Issue by marcomarinodev** [MaartenGr/BERTopic#2081]
### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Desribe the bug

Hello, first of all thanks @MaartenGr  for this amazing repo and thank you in advance for your answers.

I tried to use the zero-shot modeling, everything is amazing! But when I try to use `top_n_words` to extract more keywords, I don't see any kind of effects on my model. Looking further I've also tried to get the `c_tf_idf_` matrix as it seems to be related to the keywords, but it is set to `None`. Another try I've done was to put `top_n_words` into the representation model, but nothing happened.

Then as a last chance I tried to execute `update_topics` with the same arguments of when I fitted the model, and what I got is that now I see more keywords, the topic-documents mapping looks the same apart from the fact that the scores of each keyword is slightly changed and I loose the initial predefined labels on the topics. Another thing I noticed is that after that `c_tf_idf_` is not `None` anymore.

Do you know why this happens? I tried to deeply understand what was wrong by reading the docs, but nothing in particular changed my mind.

Thank you again

### Reproduction

```python
mmi_tm = BERTopic(
  calculate_probabilities = True,
  ctfidf_model = ctfidf_model,
  vectorizer_model=vectorizer_model,
  embedding_model = embedding_model,
  min_topic_size = 10,
  zeroshot_topic_list = taxonomy_list,
  zeroshot_min_similarity = .78,
  representation_model = KeyBERTInspired(top_n_words=15),
  top_n_words = 15,
  verbose = True,
)

mmi_topics, _ = mmi_tm.fit_transform(documents=mmi_tickets, embeddings=mmi_embeddings)

mmi_tm.get_topic(0)
```
Output:
```
[['XXXXXXXX', 0.8709571361541748], # removed for privacy
 ['error', 0.866977334022522],
 ['crashed', 0.8654656410217285],
 ['exceptionwrapper', 0.8638037443161011],
 ['exception', 0.8561791181564331],
 ['crash', 0.8546050786972046],
 ['windows', 0.8416833877563477],
 ['nullreferenceexception', 0.8410743474960327],
 ['gui', 0.8371303677558899],
 ['unable', 0.8360581398010254]]
```
```python
mmi_tm.update_topics(mmi_tickets, topics=mmi_topics, top_n_words=15, ctfidf_model=ctfidf_model, vectorizer_model=vectorizer_model, representation_model=KeyBERTInspired(top_n_words=15))

mmi_tm.get_topic(0)
```
Output
```
[('XXXXXXXX', 0.8794447),
 ('crashed', 0.87344205),
 ('error', 0.86478615),
 ('crash', 0.8647055),
 ('exceptionwrapper', 0.8554398),
 ('exception', 0.8485336),
 ('windows', 0.8412366),
 ('exe', 0.83942795),
 ('uploaded', 0.83236504),
 ('unable', 0.8312841),
 ('file', 0.8306931),
 ('nullreferenceexception', 0.8297493),
 ('version', 0.82901615),
 ('gui', 0.82838726),
 ('source', 0.8267617)]
```

### BERTopic Version

0.16.2

**Comment by marcomarinodev** [comment]
Hello, does anyone have an idea on why `top_n_words` doesn't work here?

**Comment by MaartenGr** [comment]
Could you try using the main branch of BERTopic instead? Zero-shot topic modeling went through a significant change a few commits back so I'm interested to see whether you would still have that issue.

**Comment by marcomarinodev** [comment]
Hey @MaartenGr, I tried using the master branch and it indeed worked. So now you don't have to call `update_topics` to generate more keywords. Is there a planned release to include this?

**Comment by MaartenGr** [comment]
There aren't any major PRs I believe, so I might release a new minor version in the upcoming weeks.

**Comment by marcomarinodev** [comment]
Alright, thanks Maarten!
